frontend/src/metabase/entities/data-apps/utils.unit.spec.ts

```diff
@@ -42,4 +43,52 @@ describe("data app utils", () => {
       });
     });
   });
+
+  describe("getParentDataAppPageId", () => {
+    const navItems: DataAppNavItem[] = [
+      { page_id: 1 },
+      { page_id: 2, indent: 1 },
+      { page_id: 3, indent: 0 },
+      { page_id: 4, indent: 1 },
+      { page_id: 5, indent: 1, hidden: true },
+      { page_id: 6, indent: 2 },
+    ];
+
+    it("returns null if can't find page", () => {
+      expect(
+        getParentDataAppPageId(2, [
+          { page_id: 1, indent: 1 },
+          { page_id: 2, indent: 1 },
+        ]),
+      ).toBeNull();
+    });
+
+    it("returns null if page list is empty", () => {
+      expect(getParentDataAppPageId(2, [])).toBeNull();
+    });
+
+    it("returns null for top-level page with undefined indent", () => {
+      expect(getParentDataAppPageId(1, navItems)).toBeNull();
+    });
+
+    it("returns null for top-level page with 0 indent", () => {
+      expect(getParentDataAppPageId(3, navItems)).toBeNull();
+    });
+
+    it("returns ID of the closest top-level page when indent is 1", () => {
+      expect(getParentDataAppPageId(2, navItems)).toBe(1);
+    });
+
+    it("returns ID of the closest parent page by indent for visible pages", () => {
+      expect(getParentDataAppPageId(4, navItems)).toBe(3);
+    });
+
+    it("returns ID of the closest parent page by indent for hidden pages", () => {
+      expect(getParentDataAppPageId(5, navItems)).toBe(3);
+    });
+
+    it("skips hidden pages when looking for a parent", () => {
+      expect(getParentDataAppPageId(6, navItems)).toBe(4);
```

Comment:
this strikes me as not the right solution - this makes it look like 6 is a child of 4, when it was meant to be a child of 5. it doesn't seem like parents should be allowed to be hidden - either that or we should hide their children?

Reply:
Good call btw. The function is optimized for figuring out what to highlight, but I see now that it can get confusing and counter-intuitive. Let me sync with Bruno on that